### DISCUSSION TITLE
Install a package in a workspace not the workspace root

### DISCUSSION THREAD

**Discussion by jcalfee** [pnpm/pnpm#6601] [General]
How do I install a package in the workspace?  Not the root?

They guessed right, "_which might not be what you want_".  Strange that it only goes into what I might not want but does not say anything about adding the package there like the command implies: **`pnpm add <package>`**

This shell has `./package.json` and `../pnpm-workspace.yaml` ..

>$ pnpm add bootstrap-vue
> ERR_PNPM_ADDING_TO_ROOT  Running this command will add the dependency to the workspace root, which might not be what you want - if you really meant it, make it explicit by running this command again with the -w flag (or --workspace-root). If you don't want to see this warning anymore, you may set the ignore-workspace-root-check setting to true.

>$ pnpm add -P bootstrap-vue
> ERR_PNPM_ADDING_TO_ROOT  Running this command will add the dependency to the workspace root, which might not be what you want - if you really meant it, make it explicit by running this command again with the -w flag (or --workspace-root). If you don't want to see this warning anymore, you may set the ignore-workspace-root-check setting to true.

I guess I'm not the only one: https://stackoverflow.com/q/71054629/766233

It seams undocumented in my view, "filter" is not something I would think to go into.  I would appreciate if we could know why this command does not work, pnpm is such a great product it is interesting that you get stopped pretty early on like this.  In the mean time, I'm happy to edit package.json to install packages because pnpm is worth it.

**Comment by jcalfee** [comment]
[Filtering](https://pnpm.io/filtering) does not seam to find my project, even though `pnpm install` found and installed it.
```bash
workspace/project$ pnpm add bootstrap-vue --filter project
No projects matched the filters in "/home/user/workspace/project"
```

**Reply by jcalfee** [reply]
Even though the `package.json` name matched ("project" above), it still did not match.  I tried `./` prefixing and some path tip to surround by curly brackets `{project}`.  Finally, I figured out I had to step up one directory and run the command above to get that to work.  Took me about an hour to figure out how to install a package.  Thanks, pnpm is still worth it.  Just this is a bit over the top for the primary feature of a package manager.  Simply tweak the message is what I suggest to save everyone an hour.  Thank you.

**Reply by ron2015schmitt** [reply]
same

**Comment by JxJPu9bD** [comment]
首先你应该展示你的目录结构和package.json, 
其次 --filter是过滤的包名, 你应该确认你的包名是否是 project
https://pnpm.io/zh/filtering

**Comment by JxJPu9bD** [comment]
对于你的问题, 最快的方法是你进入到project目录内再运行 
pnpm add bootstrap-vue
